zebra-state/src/service.rs

```diff
@@ -229,6 +229,31 @@ impl StateService {
             relevant_chain,
         )?;
 
+        // Reject double-spends of nullifers:
+        // - one from this block, and the other already committed to the finalized state.
+        //
+        // (Duplicate non-finalized nullifiers are rejected during the chain update.)
+        //
+        // "A nullifier MUST NOT repeat either within a transaction,
+        // or across transactions in a valid blockchain.
+        // Sprout and Sapling and Orchard nullifiers are considered disjoint,
+        // even if they have the same bit pattern."
+        //
+        // https://zips.z.cash/protocol/protocol.pdf#nullifierset
```

Comment:
Do you think I should just delete this consensus rule, and go with the shorter one?

Reply:
I ended up putting the long consensus rule at the top of the `nullifier` module, and then quoting the short or specific consensus rules for each function. See ef1d2fb4.